pkg/framework/machinesets.go

```diff
@@ -60,8 +66,29 @@ func BuildMachineSetParams(ctx context.Context, client runtimeclient.Client, rep
 	workers, err := GetWorkerMachineSets(ctx, client)
 	Expect(err).ToNot(HaveOccurred())
 
-	providerSpec := workers[0].Spec.Template.Spec.ProviderSpec.DeepCopy()
-	clusterName := workers[0].Spec.Template.Labels[ClusterKey]
+	worker := workers[0]
+
+	var arch string
+
+	if arch, err = getArchitectureFromMachineSetNodes(ctx, client, worker); err != nil {
+		klog.Warningf("error getting architecture from the machine set %s's nodes - the capacity annotation will be tried: %v", worker.Name, err)
+	}
+	// Find the first machine set that controls non-amd64 machines, if any.
```

Comment:
will we ever have a scenario with more than 2 architectures in the same cluster?

Reply:
we don't do it in the CI yet